lints/etsi/lint_qcstatem_psd2_policy_mandatory.go

```diff
@@ -0,0 +1,80 @@
+	}
+	_, isQscdStmtPresent := util.IsQcStatemPresent(c, &util.IdEtsiQcsQcSSCD)
+	if util.HasCertPolicy(c, policy) && mustHaveQcStmtFour && !isQscdStmtPresent {
+		return "EU Qualified Certificate has policy indicating private key storage in QSCD, but the corresponding QC Statement 4 is missing (ETSI EN 319 412-4: \"Policy identifiers included in the certificate policies extension of EU Qualified Certificates shall be consistent with the EU Qualified Certificate Statements\")."
+	}
+	if util.HasCertPolicy(c, policy) && !mustHaveQcStmtFour && isQscdStmtPresent {
+		return "EU Qualified Certificate does not feature the policy indicating private key storage in QSCD, but the corresponding QC Statement 4, indicating this, is present (ETSI EN 319 412-4: \"Policy identifiers included in the certificate policies extension of EU Qualified Certificates shall be consistent with the EU Qualified Certificate Statements\")."
+	}
+	return ""
+}
+
+func (l *qcStatemPsd2PolicyMandatory) Execute(c *x509.Certificate) *lint.LintResult {
+	errStr := policyAndTypeAreConsistent(c, util.IdEtsiPolicyQcpWeb, "QCP-w", util.IdEtsiQcsQctWeb, "Web", false)
+	util.AppendToStringSemicolonDelim(&errStr, policyAndTypeAreConsistent(c, util.IdEtsiPolicyQcpNatural, "QCP-n", util.IdEtsiQcsQctEsign, "eSign", false))
+	util.AppendToStringSemicolonDelim(&errStr, policyAndTypeAreConsistent(c, util.IdEtsiPolicyQcpNaturalQscd, "QCP-n-qscd", util.IdEtsiQcsQctEsign, "eSign", true))
+	util.AppendToStringSemicolonDelim(&errStr, policyAndTypeAreConsistent(c, util.IdEtsiPolicyQcpLegal, "QCP-l", util.IdEtsiQcsQctEseal, "eSeal", false))
+	util.AppendToStringSemicolonDelim(&errStr, policyAndTypeAreConsistent(c, util.IdEtsiPolicyQcpLegalQscd, "QCP-l-qscd", util.IdEtsiQcsQctEseal, "eSeal", true))
+	if errStr != "" {
+		return &lint.LintResult{Status: lint.Error, Details: errStr}
+	}
+	_, psd2Present := util.IsQcStatemPresent(c, &util.IdEtsiPsd2Statem)
+	_, QcTypePresent := util.IsQcStatemPresent(c, &util.IdEtsiQcsQcType)
+	if psd2Present && !QcTypePresent {
+		return &lint.LintResult{Status: lint.Error, Details: "EU Qualified Certificate has PSD2 QcStatement but not the mandatory QcType Statement (ETSI EN 319 412-5: \"When the certificate is issued in accordance with Annex III or Annex IV of Regulation (EU) No 910/2014 [i.8], this statement shall be present\")."}
+	}
+	if util.HasCertAnyEtsiQcStatement(c) && len(c.PolicyIdentifiers) == 0 {
+		return &lint.LintResult{Status: lint.Error, Details: "EU Qualified Certificate must at least contain one policy identifier (ETSI EN 319 411-2: GEN-6.6.1-05)"}
+	}
+	return &lint.LintResult{Status: lint.Pass}
+}
+
+func init() {
+	lint.RegisterLint(&lint.Lint{
+		Name:          "e_qcstatem_psd2_policy_mandatory",
+		Description:   "Tests various requirements regarding policy identifiers of EU Qualified Certificates",
+		Citation:      "ETSI EN 319 412-4, ETSI EN 319 412-5, ETSI EN 319 411-2",
```

Comment:
Let's reference specific sections or normative requirements here:
```suggestion
		Citation:      "ETSI EN 319 411-2: GEN-6.6.1-05, ETSI EN 319 412-4: Clause 4.3, ETSI EN 319 412-5: Clause 5",
```